src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,566 @@
+            sourceSymbolValidator: verify, symbolValidator: verifyMetadata);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyMethodIL("C", "M", """
+            .method public hidebysig 
+            	instance void M (
+            		method void *(int32& modreq([netstandard]System.Runtime.InteropServices.InAttribute)) p
+            	) cil managed 
+            {
+            	// Method begins at RVA 0x2067
+            	// Code size 1 (0x1)
+            	.maxstack 8
+            	IL_0000: ret
+            } // end of method C::M
+            """);
+
+        static void verify(ModuleSymbol m)
+        {
+            var p = m.GlobalNamespace.GetMember<MethodSymbol>("C.M").Parameters.Single();
+            var ptr = (FunctionPointerTypeSymbol)p.Type;
+            Assert.Equal(RefKind.RefReadOnlyParameter, ptr.Signature.Parameters.Single().RefKind);
+            Assert.Null(m.GlobalNamespace.GetMember<NamedTypeSymbol>(RequiresLocationAttributeQualifiedName));
+        }
+
+        static void verifyMetadata(ModuleSymbol m)
+        {
+            var p = m.GlobalNamespace.GetMember<MethodSymbol>("C.M").Parameters.Single();
+            var ptr = (FunctionPointerTypeSymbol)p.Type;
+            Assert.Equal(RefKind.In, ptr.Signature.Parameters.Single().RefKind);
+            Assert.Null(m.GlobalNamespace.GetMember<NamedTypeSymbol>(RequiresLocationAttributeQualifiedName));
+        }
+    }
+
+    [Fact]
+    public void AttributeIL()
+    {
+        var source = """
+            class C
+            {
+                public void M(ref readonly int p) { }
+            }
+            """;
+        var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyTypeIL(RequiresLocationAttributeName, """
+            .class private auto ansi sealed beforefieldinit System.Runtime.CompilerServices.RequiresLocationAttribute
+                extends [netstandard]System.Attribute
+            {
+                .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+                	01 00 00 00
+                )
+                .custom instance void Microsoft.CodeAnalysis.EmbeddedAttribute::.ctor() = (
+                	01 00 00 00
+                )
+                // Methods
+                .method public hidebysig specialname rtspecialname 
+                	instance void .ctor () cil managed 
+                {
+                	// Method begins at RVA 0x2050
+                	// Code size 7 (0x7)
+                	.maxstack 8
+                	IL_0000: ldarg.0
+                	IL_0001: call instance void [netstandard]System.Attribute::.ctor()
+                	IL_0006: ret
+                } // end of method RequiresLocationAttribute::.ctor
+            } // end of class System.Runtime.CompilerServices.RequiresLocationAttribute
+            """);
+    }
+}
```

Comment:
It feels like we should test:
- how we figure out RefKind from source for some error cases involving modifiers
- How we import some unconventional metadata for parameters, including for return parameters.